content/learn/book/ecs/resources/_index.md

```diff
@@ -6,12 +6,294 @@ page_template = "book-section.html"
+You might want to use resources for:
 
-TODO: Show how to create and access Resources
+- storing events
+- holding configuration data
+- storing simple global game state that you only need a single copy of (like a game's store)
+- interoperating with other libraries
+- providing faster or supplementary look-up data structures for entities (such as indexes or graphs)
+- storing reference-counted handles to assets (so then they're not unloaded when the last entity using it is despawned)
 
-TODO: Show how to modify resources
+[`Res`]: https://docs.rs/bevy/latest/bevy/ecs/system/struct.Res.html
+[`ResMut`]: https://docs.rs/bevy/latest/bevy/ecs/system/struct.ResMut.html
 
-TODO: Compare and contrast with `query.single()`
+## Creating resources
+
+Unlike components, resources do not need their own trait implementation: you can use any new or existing [`'static`] Rust type as a resource (if it is not [`Send + Sync`], you'll need a [`NonSend`] resource instead).
+
+Like entities and their component data, resources are stored in your [`App`]'s [`World`] struct.
+Resources are typically added statically, via [`insert_resouce`] or [`init_resource`].
+
+[`insert_resource`] is used when you want to set the value of a resource manually, while [`init_resource`] is used when you want to automatically initialize the resources value using the [`Default`] or [`FromWorld`] trait.
+
+```rust
+use bevy::prelude::*;
+
+// Resources can be tuple structs
+// Default can be derived for many simple resources,
+// with the default value of most numeric types being 0
+#[derive(Default)]
+struct Score(u64);
+
+// Resources can be ordinary namedstructs
+struct PlayerSupplies {
+ gold: u64,
+ wood: u64,
+}
+
+// The Default trait can be manually implemented to control initial values
+impl Default for PlayerSupplies {
+ fn default() -> Self {
+  PlayerSupplies {
+   gold: 400,
+   wood: 200,
+  }
+ }
+}
+
+// Resources can be enums
+enum Turn {
+ Allied,
+ Enemy
+}
+
+fn main(){
+ // Resources are typically inserted using AppBuilder methods
+ App::build()
+ // Uses the default() value provided by the derived Default trait
+ .init_resource::<Score>()
+ // Uses the default() value provided by the manual impl of the Default trait
+ .init_resource::<PlayerSupplies>()
+ // Uses the specific manual value of Turn::Allied to insert a resource of type Turn
+ .insert_resource(Turn::Allied)
+ // Sets the value of the standard Bevy resource `WindowDescriptor`,
+ // leaving the unspecified fields as their default value
+ .insert_resource(WindowDescriptor {
+  title: "I am a window!".to_string(),
+  width: 500.,
+  height: 300.,
+  vsync: true,
+  ..Default::default()
+ })
+ .add_plugins(DefaultPlugins)
+ .run()
+}
+```
+
+In rare cases, you may need to add a resource later, once other parts of the world exist to ensure proper initialization.
+For that, we can use the equivalent methods on [`Commands`].
+You can add, overwrite and even [`remove_resource`] dynamically in this way.
```

Comment:
There should definitely be a warning here about removing required resources, as that will cause systems that expect them to be there to panic. Perhaps a note that interfaces around resources should all have a similar expectation of whether the resource is present or not.